feat: Add RevokeToken tool for removing saved payment methods

himanshu-rzp opened the pull request:
## Description

* Add RevokeToken function in tokens.go with customer_id and token_id parameters
* Register revoke_token tool in payments toolset as write tool
* Add comprehensive unit tests covering positive, negative, and edge cases
* Update README.md with new tool documentation and correct API link
* All tests pass and linter checks clean

## Related Issues

## Type of Change

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
* [ ] Refactoring (no functional changes, code improvements only)
* [ ] Documentation update
* [ ] Performance improvement

## Testing

* [x] Manual testing
* [x] Added unit tests
* [ ] Added integration tests (if applicable)
* [x] All tests pass locally

## Checklist

* [x] I have followed the code style of this project
* [x] I have added comments to code where necessary, particularly in hard-to-understand areas
* [x] I have updated the documentation where necessary
* [x] I have verified that my changes do not introduce new warnings or errors
* [x] I have checked for and resolved any merge conflicts
* [x] I have considered the performance implications of my changes

## Additional Information

The revoke_token tool allows users to cancel saved payment methods (tokens) for customers. This tool uses the Razorpay Cancel Token API endpoint with PUT method to properly revoke tokens from the payment system.

jating06 commented on pkg/razorpay/tokens_test.go:
```diff
@@ -409,3 +409,255 @@ func Test_FetchSavedPaymentMethods_ClientContextScenarios(t *testing.T) {
+		request := mcpgo.CallToolRequest{
+			Arguments: map[string]interface{}{
+				"customer_id": "cust_1Aa00000000003",
+				"token_id":    "token_EhYXHrLsJdwRhM",
```
change token with some dummy value

himanshu-rzp committed:
feat: add RevokeToken tool for canceling saved payment methods

- Add RevokeToken tool to revoke saved payment methods for customers
- Implement proper API endpoint: PUT /v1/customers/{customer_id}/tokens/{token_id}/cancel
- Add comprehensive unit tests with positive, negative, and edge cases
- Register tool in payments toolset for write operations
- Update README.md with new tool documentation and API reference
- Follow Razorpay Cancel Token API specification for UPI OTM collect tokens
